### ISSUE TITLE
Dependencies and Suggestion(s)

### ISSUE THREAD

**Issue by BakProject** [FlyingPikachu/Quests#193]
# Quests

Hey Pikachu, hopefully you are reading this!

I'm sure you get lots of messages and post about issues on you plugin. But maybe you will read this one. I have started using your plugin and I can't figure out what quest points are used for (EXCEPT FOR REQUIREMENTS). If there is something else please reply back here if you can. If there isn't anyother thing to use them for, I think that you could implement a level system allowing you to rank-up through finishing quests and gaining quest points!

_**ChestCommands:**_
I think this plugin might be good to have a dependency so that we can make our own GUI's for the quests. I have made a GUI with this plugin and disabling the GUI Toggle and using CitizensCMD to open the menu and have clickable commands. It doesn't accept them right away players still have to type "Yes" or "No" in chat. Only problem is that if you have a stage with NPC talking the menu pops up. 

_**Level-Up System:**_
This section would have to be done with a config file unless you think of a way to do it in-game. Or another plugin(I would definitely download the level-up plugin). You use set up different ranks using quest points such as, "Quester", "Master", "Adventurer", etc.. You could hav ea dependency of PEX using the 'rank' system.

This plugin is very good I like the in-game setup, it's so easy and amazing! Anyways thank you for reading this I know time is valuable and I appreciate your time. Have a good one.

Sorry for the lengthy 
    - BakProject (MCForest.net)

**Comment by FlyingPikachu** [comment]
Afaik quest points are only used for requirements, but I recall it being suggested their functionality could be expanded on for other purposes. All of the ideas you've posted seem interesting, except for the Crafting quests as that has been explored already (see #14).

Marking as Enchancement for future consideration

**Comment by BakProject** [comment]
A player of the server actually requested a level up system and thought I could request one. Or atleast suggest to see what could happen. But my main one is ChestCommands. That would be AMAZING!

**Comment by BakProject** [comment]
Is there any way I could contact you? Like a discord maybe?

**Comment by johnsoncui02** [comment]
I would recommend removing the yes or no confirmation from the quest in order to make chestcommands to be compatible with quests nicely. -

**Comment by FlyingPikachu** [comment]
So it’s my understanding that the only change needed to play nice with ChestCommands is to disable the Yes / No confirmation? If so, adding a config option to disable that should be pretty simple. Assuming that’s the case, expect it in the next release.

**Comment by FlyingPikachu** [comment]
Dev build - http://ci.ac3-servers.eu/job/Quests/230/

Option in config.yml available as `ask-confirmation` enabled by default. Note that users will not be able to see a quest description before taking the quest unless you're using GUI.

This build is also the big jump to the new Crowdin translation system, so if you have a customized language file, you'll have to work it into the new arrangement. Documentation is forthcoming.

**Comment by johnsoncui02** [comment]
![image](https://user-images.githubusercontent.com/23487437/34654372-9df96d50-f3af-11e7-9199-5c3489e9f35e.png)

After changing the quest plugin to build #230, while following your suggested guide here's what it shows.

**Comment by FlyingPikachu** [comment]
@johnsoncui02 Oops, I only tested it using NPCs. If you have Citizens installed, could you test whether accepting a quest through an NPC works as expected? I’ll take a look tonight.

**Comment by johnsoncui02** [comment]
oh, if you were concerned about right click event with npc in GUI it works without a requirement for yes/no already. I was suggesting the actual command /quest take (questname) to be removed from the yes/no from showing.

**Comment by BakProject** [comment]
Yes the command should able to be toggleable. I am using ChestCommands GUI. And I am not using the plugins custom GUI and so I have to run the quest take through command. I have also downloaded the new update and tested this, so it didn't really fix much for me. Maybe a  command toggle? This is what I'm running with I. ChestCommand.

Example:

menu-background-15:
  COMMAND: 'quests take Obsidian Grabber'
  NAME: '&cObsidian Grabber'
  ID: '49'
  POSITION-X: 7
  POSITION-Y: 2
  LORE:
  - ''
  - '&cRequirements:'
  - '&b&l(&615 Quest Points&b&l)'
  - '&b&l(&6Stone Miner Quest&b&l)'
  - '&b&l(&6Time Attack Quest&b&l)'

It would be very helpful if I could just have the player run that command with only clicking on the Item Command in chest commands and completely skip the yes or no interaction.

**Comment by FlyingPikachu** [comment]
Dev build - http://ci.ac3-servers.eu/job/Quests/231/

This build will also remove the Yes/No when `ask-confirmation` is disabled and _/quests take_ is used. I've tested that it's working, but lemme know if I missed anything. Once again, bear in mind the player will not receive a description of the quest before accepting!

**Comment by johnsoncui02** [comment]
@BakProject somewhat finished my organizing in wordings. let me know if you need a copy to slap the organization on.

![image](https://user-images.githubusercontent.com/23487437/34664110-161f9d40-f40f-11e7-96aa-dab6ee112c00.png)

@FlyingPikachu This is my current way to tell people to quit a quest. as you can see the example below. if it's possible to create a command to drop all quests at once. It would solve alot of problems. Just thought this might provide a better example.
![image](https://user-images.githubusercontent.com/23487437/34664121-2499805c-f40f-11e7-8530-a73d91afbc53.png)

**Comment by johnsoncui02** [comment]
@BakProject are you experiencing this afterwards also? I tried to remove the npc and adjust all the ID. after server restarts it still gave me this. Any hotfix for it?
![image](https://user-images.githubusercontent.com/23487437/34685121-32e9f716-f45c-11e7-925d-75a355190eb4.png)

**Comment by BakProject** [comment]
@johnsoncui02 I am not getting that error. I actually changed mine to a shortcut to the menu which I tell the players to use. So mine says "Use /q for questing on the go... Etc.." everything works fine for me other then the Yes / No

**Comment by johnsoncui02** [comment]
other then the yes/no? did you check the latest update? It fixes that problem @BakProject pika made a change to it.

**Comment by BakProject** [comment]
I have downloaded the new update but have not restarted my server yet! But when I get home I will restart when I get home. And I'll let you know @johnsoncui02 Maybe even a picture of my gui and chat message.

**Comment by johnsoncui02** [comment]
so i figured the issue with the misleading message. It registers the npc as a Quest NPC due to adding the npc into the talk-to event. I do this to make the NPC seem more useful in a way and more interactive with the user rather than just taking the quest and finish it and pretend the npc is no use after the initial quest finishing. do you think Mr.Pika is still paying attention to this thread or should i make a suggestion for him to see how he feels about it?

![image](https://user-images.githubusercontent.com/23487437/34706313-a8020066-f4bb-11e7-9412-ec5d2e5f7e1c.png)

**Comment by BakProject** [comment]
@johnsoncui02 he gets emails just like you do. But I would create a new ticket. I have made one about chat triggers for events and am waiting for a tag.

**Comment by Felixx61** [comment]
@johnsoncui02 did you create that GUI externally via CC or internally with Quests?

I'm looking to add on a GUI system for Quests and am trying to think of the most efficient way of doing so.

**Comment by johnsoncui02** [comment]
@Felixx61 i recommend using chestcommands. the downside is that there's no clear way to actually have a easy quit quest function say. /quests quit all or things of sorts. This is my finished product however.

![image](https://user-images.githubusercontent.com/23487437/34922983-ec83ee22-f94b-11e7-83a5-3729b093ff80.png)

**Comment by BakProject** [comment]
@Felixx61i have a gui icon with chestcommands and have a icon that runs a command through TitleManager you can have the console send them a player broadcast  to show them how to quit.

**Comment by johnsoncui02** [comment]
or you can just have this on the side.
![image](https://user-images.githubusercontent.com/23487437/34923509-7cf798c8-f951-11e7-8855-023606f3e5ec.png)
